crates/circuit/src/dag_circuit.rs

```diff
@@ -2944,14 +2883,13 @@ impl DAGCircuit {
             self.dag.add_edge(
                 pred.source(),
                 succ.target(),
-                Wire::Var(
-                    self.vars
-                        .find(&PyObjectAsKey::new(&contracted_var))
-                        .unwrap(),
-                ),
+                Wire::Var(self.vars.find(contracted_var).unwrap()),
             );
         }
 
+        drop(input_dag_var_set);
+        drop(node_vars);
```

Comment:
Were you worried about the memory overhead here? These would get dropped pretty soon after automatically when the function exits.

Reply:
At the very least, we actually _need_ to drop `input_dag_var_set` since it holds references to variables in the DAG, and we need to mutate the DAG later in this function.

I don't remember if `node_vars` needs to be dropped for the same reason, but it also holds references to things in `self.vars`.

Normally, I like the idea of dropping as a result of leaving a lexical scope, but that didn't seem worth the extra indentation given how many lines we'd need within it.

Reply:
That's fine but then can you put a comment here explaining the rationale. Seeing manual `drop()` calls is a bit uncommon and I think warrants some explanation.